cli/command/image/pull.go

```diff
@@ -74,7 +74,7 @@ func runPull(dockerCli *command.DockerCli, opts pullOptions) error {
 		err = imagePullPrivileged(ctx, dockerCli, authConfig, distributionRef.String(), requestPrivilege, opts.all)
 	}
 	if err != nil {
-		if strings.Contains(err.Error(), "target is plugin") {
+		if strings.Contains(err.Error(), "MediaType is plugin") {
 			return errors.New(err.Error() + " - Use `docker plugin install`")
```

Comment:
Perhaps the CLI should strip the mediatype here (and in the plugin case) to make the user-visible error less verbose.

I agree that having the mediatype in the general case is useful.